File: farmaciaX/wwwroot/js/productos.js
const app = Vue.createApp({
    data() {
        return {
            productos: [],
            productosFiltrados: [],
            filtro: '',
            modoEdicion: false,
            producto: {
                id: 0,
                nombre: '',
                tipo: '',
                precio: 0,
                cantidad_Stock: 0,
                laboratorio: '',
                requiere_Receta: false,
                fecha_Vencimiento: '',
                activo: true
            }
        };
    },
    methods: {
        async cargarProductos() {
            const res = await fetch('/api/Productos/GetProductos');
            this.productos = await res.json();
            this.productosFiltrados = this.productos;
        },
        filtrarProductos() {
            const texto = this.filtro.toLowerCase();
            this.productosFiltrados = this.productos.filter(p =>
                p.nombre.toLowerCase().includes(texto)
            );
        },
        editarProducto(p) {
            this.producto = { ...p };
            this.modoEdicion = true;
        },
        resetearFormulario() {
            this.producto = {
                id: 0,
                nombre: '',
                tipo: '',
                precio: 0,
                cantidad_Stock: 0,
                laboratorio: '',
                requiere_Receta: false,
                fecha_Vencimiento: '',
                activo: true
            };
            this.modoEdicion = false;
        },
        async guardarProducto() {
            const metodo = this.producto.id ? 'PUT' : 'POST';
            const url = this.producto.id
                ? `/api/Productos/${this.producto.id}`
                : '/api/Productos';

            const res = await fetch(url, {
                method: metodo,
                headers: { 'Content-Type': 'application/json' },
                body: JSON.stringify(this.producto)
            });

            if (!res.ok) {
                alert('Error al guardar');
                return;
            }

            await this.cargarProductos();
            this.resetearFormulario();
        },
        async darDeBaja(p) {
            if (!confirm(`¿Dar de baja a ${p.nombre}?`)) return;

            const baja = { ...p, activo: false };
            const res = await fetch(`/api/Productos/${p.id}`, {
                method: 'PUT',
                headers: { 'Content-Type': 'application/json' },
                body: JSON.stringify(baja)
            });

            if (!res.ok) {
                alert('Error al dar de baja');
                return;
            }

            await this.cargarProductos();
        }
    },
    mounted() {
        this.cargarProductos();
    }
});

app.mount('#app');

File: farmaciaX/Models/Productos.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace farmaciaX.Models;
public class Productos
{
    public int Id { get; set; }
    public string Nombre { get; set; }
    public string Tipo { get; set; }
    
    [Column(TypeName = "decimal(18,2)")]
    public decimal Precio { get; set; }
    public int Cantidad_Stock { get; set; }
    public bool Requiere_Receta { get; set; }
    public string Laboratorio { get; set; }
    public DateTime Fecha_Vencimiento { get; set; }
    public bool Activo { get; set; } = true;

    public override string ToString()
    {
        return $"Id: {Id}, Nombre: {Nombre}, Tipo: {Tipo}, Precio: {Precio}, Cantidad_Stock: {Cantidad_Stock}, Requiere_Receta: {Requiere_Receta}, Laboratorio: {Laboratorio}, Fecha_Vencimiento: {Fecha_Vencimiento}";
    }
}
